src/components/DnsSettings.tsx

```diff
@@ -1,313 +1,239 @@
-import React, { PureComponent } from 'react';
+import React, { FC, ChangeEvent, useReducer, useEffect, useState } from 'react';
 import { css } from 'emotion';
 import { Collapse, Container, HorizontalGroup, Field, Select, MultiSelect, Input } from '@grafana/ui';
 import { SelectableValue } from '@grafana/data';
-import { IpVersion, Settings, DnsSettings, DnsProtocol, DnsRecordType, DNSRRValidator } from 'types';
+import { IpVersion, Settings, DnsSettings, DnsProtocol, DnsRecordType, DNSRRValidator, DnsResponseCodes } from 'types';
 import { IpOptions } from './utils';
-import ListInput from './ListInput';
+import DnsValidatorForm from './DnsValidatorForm';
+import { enumToStringArray } from 'utils';
 
 interface Props {
   settings: Settings;
   isEditor: boolean;
   onUpdate: (settings: Settings) => void;
 }
 
-interface State extends DnsSettings {
-  showAdvanced: boolean;
-  showValidation: boolean;
+const RESPONSE_CODES = enumToStringArray(DnsResponseCodes).map(responseCode => ({
```

Comment:
Consider moving all these constants to a separate file so they don't add extra clutter.